test/language/languageUnderstandingTest.js

```diff
@@ -1,40 +1,276 @@
 const cognitive = require('../../src/index.js');
 const config = require('../config.js');
+const promiseRetry = require('promise-retry');
 
-describe('Language understanding (LUIS)', () => {
+/*
+
+You only need to set the apiKey for these tests. 
+
+The app created is the prebuilt "Web" app. Creation of the app
+will fail if your LUIS account already has an app with the 
+name "Web". 
+
+Each time the app is created, its apiKey is displayed along
+with the count of training status calls. Usually, it takes more 
+than 1 call to return successfully trained status. This test will 
+not try more than retryCount times and wait retryInterval between tries. 
+
+*/
+
+describe.only('Language understanding (LUIS)', () => {
+
+    const retryCount = 10;
+    const retryInterval = 1000;
 
     const client = new cognitive.languageUnderstanding({
         apiKey: config.languageUnderstanding.apiKey,
         appID: config.languageUnderstanding.appID,
         versionID: config.languageUnderstanding.versionID,
         endpoint: config.languageUnderstanding.endpoint
     });
+    var isTrained = (trainingStatus) => {
+        var untrainedModels = trainingStatus.filter(model => {
+            if( model.details.status === 'Fail' || model.details.status === 'InProgress') return model; 
+        });
+        return (untrainedModels.length===0) ? true : false;
+    }
+    var waitUntilTrained = (client) => {
+
+        return promiseRetry((retry, number) => {
+        
+            return client.getTrainStatus()
+            .then(response => {
+                // 2xx http response 
+                let trained = client.isTrained(response);
+                console.log(number + " trained = " + trained);
+                if (!trained)retry("not trained");
+                return response;
+            })
+            .catch((err) => {
+                if (err.code === 'ETIMEDOUT') {
+                    retry(err);
+                }
+                throw err;
+            });
+        });  
+    }  
+    var createTrainPublishApp = () => {
+
+        client.isTrained = isTrained;
+        let body = {
+            "domainName": "Web", 
+            "culture": "en-us"
+        };
+        return client.addPrebuiltDomain(body).then(results =>{
+            client.appID = results.substring(results.length-36, results.length);
```

Comment:
You should move this logic to `addPrebuiltDomain` and then change the docs to `@returns {Promise.<string>}`. Same thing for `importApp()`

Reply:
I don't want to do that just yet until I talk with the PM to see what is going on with API. The APIs that create apps changed from just returning the App Id to returning a somewhat meaningless url. I want to make sure it isn't changing again and that the test checks the returned value in a meaningful way.

Reply:
Ok!